### PR TITLE
feat: use branch prerelease property for detection

### LINKED ISSUE #864: Improvement of prerelease check
As of https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md?plain=1#L91 the property prerelease can be used to configure a prerelease. If you set this property to false the prerelease is still created if the other conditions are met. With this PR you can set prerelease to false and no prerelease is created.

Fix: https://github.com/semantic-release/github/pull/863

### PR DESCRIPTION
**Pull request by klopfdreh**
For the description have a look into the issue: https://github.com/semantic-release/github/issues/864

For a showcase have a look at the video provided by @babblebey in the comment: https://github.com/semantic-release/github/pull/863#issuecomment-2192578725 

Fixes #864

### CODE REVIEW

File: lib/is-prerelease.js
Comment (babblebey) [inline lib/is-prerelease.js:8]:
```suggestion
    if (typeof prerelease === "boolean" && prerelease === false) {
        return false;
    }
    return type === "prerelease"
        || (type === "release" && !main)
        || typeof prerelease == "string"
        || prerelease === true
```

I don't think `prerelease` is ever `true`, its either `false` or `[git_ref]` 😉

**Review by babblebey — CHANGES_REQUESTED** [review]
Great stuff @klopfdreh,

Thanks for the test, super valuable 😉, kindly address the following..

- The suggested changes 
- **The Test failing in CI**: Do address the lint issues, by running the command `npm run lint:prettier:fix` on your dev branch... then run the tests again see whether it passes

I think this PR is almost good to go 👍 

**On other note:**

I have edited your PR description to properly link your Issue to the PR, this style of linking Issues to PR help automatically closes an Issue once the linked PR gets merged, kindly take note of this approach 😉

File: lib/is-prerelease.js
Comment (klopfdreh) [reply lib/is-prerelease.js]: The problem here is that prerelease can either be `true` or `false` / `undefined` or a string value as of the definition I read in semantic release. That's the reason why I don't want to consider to only return prerelease if it is undefined but only if it is `prerelease === true`. If prerelease is undefined it would return false - right?

**Comment by klopfdreh** [comment]
> Great stuff @klopfdreh,
> 
> Thanks for the test, super valuable 😉, kindly address the following..
> 
> * The suggested changes
> * **The Test failing in CI**: Do address the lint issues, by running the command `npm run lint:prettier:fix` on your dev branch... then run the tests again see whether it passes
> 
> I think this PR is almost good to go 👍
> 
> **On other note:**
> 
> I have edited your PR description to properly link your Issue to the PR, this style of linking Issues to PR help automatically closes an Issue once the linked PR gets merged, kindly take note of this approach 😉

Thanks a lot for all the hints. I force pushed the changes after using `npm run lint:prettier:fix` - I think all lint issues should be fixed, now.

File: lib/is-prerelease.js
Comment (babblebey) [reply lib/is-prerelease.js]: Good thought, It's okay for addressing the Edge case... We can always iterate over it later 😉

**Comment by babblebey** [comment]
> Thanks a lot for all the hints. I force pushed the changes after using `npm run lint:prettier:fix` - I think all lint issues should be fixed, now.

Great stuff @klopfdreh, 

This is set... but d'you know one thing that can make a PR look even good!? 😉 A screen recording of a demo where you run the changes to show it really works as expected 😉, this is super valuable for Reviewers 👨‍💻 if it were to be present in the PR description❤️ 

I have tested it and below is a screencast demoing your change for the attention of @semantic-release/maintainers  🤝 

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2024.06.26-21_20_37.webm](https://github.com/semantic-release/github/assets/25631971/41a8b767-33d0-4436-a700-88e10e610e35)

**Comment by klopfdreh** [comment]
I am going to link the showcase in the PR description - thanks for the advice 👍 - next time I am going to also provide a screen recording.

File: lib/is-prerelease.js
Comment (gr2m) [inline lib/is-prerelease.js:2]: This is the same as

```suggestion
  if (prerelease === false) {
```

The triple equal operator checks if the variables are equal in both value _and_ type

**Review by gr2m — APPROVED** [review]
Looks good to me, just a tiny nit

**Comment by gr2m** [comment]
I'll let @babblebey take it from here 👍🏼

**Review by babblebey — APPROVED** [review]
Let's goooo 💪🏾

**Comment by klopfdreh** [comment]
> Let's goooo 💪🏾

💯 Let‘s gooo 😄

**Comment by babblebey** [comment]
Thank you for your contribution @klopfdreh 😁

**Pull request merged**
